AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -372,6 +372,7 @@ - (void)acquireTokenInteractively {
                                 completionBlock:^(MSALResult *result, NSError *e) {
                                   typeof(self) strongSelf = weakSelf;
                                   if (e) {
+                                    [[MSAuthTokenContext sharedInstance] setAuthToken:nil withAccountId:nil expiresOn:nil];
```

Comment:
We need that in silent sign in as this is the one used for refresh, here the method is called interactive so that might not be the flow used during refresh.
This is good to have as well but should not be enough for the reported bug.